bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/accessories/HomekitCharacteristicFactory.java

```diff
@@ -0,0 +1,452 @@
+            (callback) -> updater.subscribe(item, CARBON_MONOXIDE_LEVEL.getTag(), callback),
+            ()->  updater.unsubscribe(item, CARBON_MONOXIDE_LEVEL.getTag()));
+    }
+    private static CarbonDioxidePeakLevelCharacteristic createCarbonDioxidePeakLevelCharacteristic(final GenericItem item, HomekitAccessoryUpdater updater) {
+        return new CarbonDioxidePeakLevelCharacteristic(
+            () -> CompletableFuture.completedFuture((item !=null) ?
+                                                    item.getStateAs(DecimalType.class).doubleValue() : 0),
+            (callback) -> updater.subscribe(item, CARBON_MONOXIDE_PEAK_LEVEL.getTag(), callback),
+            ()->  updater.unsubscribe(item, CARBON_MONOXIDE_PEAK_LEVEL.getTag()));
+    }
+    private static CurrentHorizontalTiltAngleCharacteristic createCurrentHorizontalTiltAngleCharacteristic(final GenericItem item, HomekitAccessoryUpdater updater) {
+        return new CurrentHorizontalTiltAngleCharacteristic(
+            () -> CompletableFuture.completedFuture((item !=null) ?
+                                                    item.getStateAs(DecimalType.class).intValue() : 0),
+            (callback) -> updater.subscribe(item, CURRENT_HORIZONTAL_TILT_ANGLE.getTag(), callback),
+            ()->  updater.unsubscribe(item, CURRENT_HORIZONTAL_TILT_ANGLE.getTag()));
+    }
+    private static CurrentVerticalTiltAngleCharacteristic createCurrentVerticalTiltAngleCharacteristic(final GenericItem item, HomekitAccessoryUpdater updater) {
+        return new CurrentVerticalTiltAngleCharacteristic(
+            () -> CompletableFuture.completedFuture((item !=null) ?
+                                                    item.getStateAs(DecimalType.class).intValue() : 0),
+            (callback) -> updater.subscribe(item, CURRENT_VERTICAL_TILT_ANGLE.getTag(), callback),
+            ()->  updater.unsubscribe(item,  CURRENT_VERTICAL_TILT_ANGLE.getTag()));
+    }
+    private static TargetHorizontalTiltAngleCharacteristic createTargetHorizontalTiltAngleCharacteristic(final GenericItem item, HomekitAccessoryUpdater updater) {
+        return new TargetHorizontalTiltAngleCharacteristic(
+            () -> CompletableFuture.completedFuture((item !=null) ?
+                                                    item.getStateAs(DecimalType.class).intValue() : 0),
+            (angle) -> item.setState(new DecimalType(angle)),
+            (callback) -> updater.subscribe(item, TARGET_HORIZONTAL_TILT_ANGLE.getTag(), callback),
+            ()->  updater.unsubscribe(item, TARGET_HORIZONTAL_TILT_ANGLE.getTag()));
+    }
+    private static TargetVerticalTiltAngleCharacteristic createTargetVerticalTiltAngleCharacteristic(final GenericItem item, HomekitAccessoryUpdater updater) {
+        return new TargetVerticalTiltAngleCharacteristic(
+            () -> CompletableFuture.completedFuture((item !=null) ? item.getStateAs(DecimalType.class).intValue() : 0),
```

Comment:
Since you do that alot, wouldn't it make sense to have a method for that (double in a similar way)?

```
private static CompletableFuture<int> getCompletedFutureInt(GenericItem item) {
  return CompletableFuture.completedFuture(item !=null ? item.getStateAs(DecimalType.class).intValue() : 0);
}
```

```suggestion
            HomekitCharacteristicFactory::getCompletedFutureInt,
```